docs/feature/1609-add-apk-name-to-result-matrix.md

```diff
@@ -0,0 +1,102 @@
+│ OUTCOME │      MATRIX ID       │      TEST AXIS VALUE       │               TEST DETAILS                │
+├─────────┼──────────────────────┼────────────────────────────┼───────────────────────────────────────────┤
+│ failure │ matrix-b1fizzmw53qka │ NexusLowRes-29-de-portrait │ 82 test cases failed, 206 passed, 1 flaky │
+│ failure │ matrix-wq8upcrn5uoca │ NexusLowRes-29-de-portrait │ 5 test cases failed, 4 passed             │
+│ failure │ matrix-2n50loc2ze1h4 │ NexusLowRes-29-de-portrait │ 1 test cases failed, 6 passed             │
+│ success │ matrix-3eo91na6dzmd1 │ NexusLowRes-29-de-portrait │ 8 test cases passed                       │
+│ failure │ matrix-1ztkep4prcq43 │ NexusLowRes-29-de-portrait │ 1 test cases failed                       │
+│ failure │ matrix-w9wpp0jwt4yba │ NexusLowRes-29-de-portrait │ 23 test cases failed, 6 passed            │
+│ success │ matrix-30cqwb4sc1rqh │ NexusLowRes-29-de-portrait │ 17 test cases passed                      │
+│ success │ matrix-32kzu6e7a1t3h │ NexusLowRes-29-de-portrait │ 2 test cases passed                       │
+│ failure │ matrix-3dp8qqblpyav7 │ NexusLowRes-29-de-portrait │ 20 test cases failed, 12 passed           │
+│ failure │ matrix-zw52wsouk4zma │ NexusLowRes-29-de-portrait │ 1 test cases failed                       │
+│ INVALID │ matrix-36qxkc2szybc6 │                            │ NO_TEST_RUNNER_CLASS                      │
+└─────────┴──────────────────────┴────────────────────────────┴───────────────────────────────────────────┘
+```
+
+# References
+
+- [Issue #1609](https://github.com/Flank/flank/issues/1609)
+
+# Motivation
+
+Flank gives a possibility to run multiple apks in parallel within a single test run. This feature is extensively used by companies with big projects with lots of additional modules. Unfortunately, processing results in the current form is neither easy nor straightforward. The user needs to associate matrix with related apk/module which can be time-consuming and error-prone. To achieve a better result overview, flank should add apk name to the both result table and output JSON.
+
+# Goals
+
+Add an associated apk file name to the result table and output JSON.
+
+# Design
+
+`Flank` is not aware of any project, modules, etc so it can't find module names on its own, in a reasonable way, without breaking backward compatibility.
+
+On the other hand finding, the apk file name is simple and straightforward. To achieve it we need:
+1. add new field `app` to the `SavedMatrix` data class
+2. update private extension function `SavedMatrix#updateProperties` to extract apk file name from the `newMatrix`
+   * find app used during tests under the `TestMatrix` -> `TestSpecification`
+   * once we have `TestSpecyfication`, iterate over all specs (android/ios/instrumentation/etc.) and find non-null
+   * since we can run only one spec within a matrix, only one (spec) is non-null
+   * fetch `appApk` from spec (this is full `gs://` path) `TestSpecyfication#appApk`
```

Comment:
```suggestion
   * fetch `appApk` from spec (this is full `gs://` path) `TestSpecification#appApk`
```
The y in the middle here is incorrect. If the class is called this we should change it :)

Reply:
Nah, that my mistake, thanks!